src/lTerm_read_line.ml

```diff
@@ -485,6 +514,35 @@ object(self)
                 Zed_edit.insert context (Zed_rope.of_string line)
         end
 
+      | History_search_prev when S.value mode = Edition -> begin
+          let prev, next = S.value history in
+          let text = Zed_rope.to_string @@ Zed_edit.text edit in
+          let prefix = if next <> [] then last_exn next else text in
```

Comment:
It will keep filtering for the original term, but probably filtering by the edited term makes more sense. I'll fix it.